Slack Notifications blocks_template does not support booleans

### The problem

When using the notify.slack service, supplying a `blocks_template` that contains a boolean field fails with the error 

```
TypeError: Expected template to be a string
```

Specifically for me, when I added a button block that has the `emoji: true` element, generating the template fails. Making the value a string (eg `emoji: "true"`) fails because when it renders to JSON to send to Slack, it no longer meets the schema: 

```
There was a Slack API error while sending to random: SlackApiError("The request to the Slack API failed.\nThe server responded with: {'ok': False, 'error': 'invalid_blocks', 'errors': ['failed to match all allowed schemas [json-pointer:/blocks/2/elements/0/text]', 'must be a boolean [json-pointer:/blocks/2/elements/0/text/emoji]'], 'response_metadata': {'messages': ['[ERROR] failed to match all allowed schemas [json-pointer:/blocks/2/elements/0/text]', '[ERROR] must be a boolean [json-pointer:/blocks/2/elements/0/text/emoji]']}}")
```

I presume any elements that are not string leaf nodes (int, bool, etc) will fail in the same way.


### What is version of Home Assistant Core has the issue?

core-2021.7.3

### What was the last working version of Home Assistant Core?

_No response_

### What type of installation are you running?

Home Assistant OS

### Integration causing the issue

Slack

### Link to integration documentation on our website

https://www.home-assistant.io/integrations/slack

### Example YAML snippet

```yaml
alias: Test Switch Notification
description: ''
trigger:
  - platform: state
    entity_id: input_boolean.test_switch
condition: []
action:
  - service: notify.slack
    data:
      target: '#random'
      message: Test Switch
      data:
          username: Automation Tester
          icon: ":test_tube:"
          blocks_template:
            - type: section
              text:
                type: mrkdwn
                text: "Automation Test Switch {{ states('input_boolean.test_switch') }}"
            - type: divider
            - type: actions
              block_id: input_boolean.test_switch
              elements:
                - type: button
                  text:
                    type: plain_text
                    text: Turn off
                    emoji: true
                  value: "Turn off"

mode: single
```


### Anything in the logs that might be useful for us?

```txt
2021-07-28 14:38:33 ERROR (MainThread) [homeassistant.components.automation.test_switch_notification] Test Switch Notification: Error executing script. Unexpected error for call_service at pos 1: Expected template to be a string
Jul 28 19:38:33 homeassistant 7249604ba3ea[423]: Traceback (most recent call last):
Jul 28 19:38:33 homeassistant 7249604ba3ea[423]:   File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 371, in _async_step
Jul 28 19:38:33 homeassistant 7249604ba3ea[423]:     await getattr(self, handler)()
Jul 28 19:38:33 homeassistant 7249604ba3ea[423]:   File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 571, in _async_call_service_step
Jul 28 19:38:33 homeassistant 7249604ba3ea[423]:     await service_task
Jul 28 19:38:33 homeassistant 7249604ba3ea[423]:   File "/usr/src/homeassistant/homeassistant/core.py", line 1491, in async_call
Jul 28 19:38:33 homeassistant 7249604ba3ea[423]:     task.result()
Jul 28 19:38:33 homeassistant 7249604ba3ea[423]:   File "/usr/src/homeassistant/homeassistant/core.py", line 1526, in _execute_service
Jul 28 19:38:33 homeassistant 7249604ba3ea[423]:     await handler.job.target(service_call)
Jul 28 19:38:33 homeassistant 7249604ba3ea[423]:   File "/usr/src/homeassistant/homeassistant/components/notify/__init__.py", line 159, in _async_notify_message_service
Jul 28 19:38:33 homeassistant 7249604ba3ea[423]:     await self.async_send_message(**kwargs)
Jul 28 19:38:33 homeassistant 7249604ba3ea[423]:   File "/usr/src/homeassistant/homeassistant/components/slack/notify.py", line 318, in async_send_message
Jul 28 19:38:33 homeassistant 7249604ba3ea[423]:     blocks = _async_templatize_blocks(
Jul 28 19:38:33 homeassistant 7249604ba3ea[423]:   File "/usr/src/homeassistant/homeassistant/components/slack/notify.py", line 154, in _async_templatize_blocks
Jul 28 19:38:33 homeassistant 7249604ba3ea[423]:     return [_async_templatize_blocks(hass, item) for item in value]
Jul 28 19:38:33 homeassistant 7249604ba3ea[423]:   File "/usr/src/homeassistant/homeassistant/components/slack/notify.py", line 154, in <listcomp>
Jul 28 19:38:33 homeassistant 7249604ba3ea[423]:     return [_async_templatize_blocks(hass, item) for item in value]
Jul 28 19:38:33 homeassistant 7249604ba3ea[423]:   File "/usr/src/homeassistant/homeassistant/components/slack/notify.py", line 156, in _async_templatize_blocks
Jul 28 19:38:33 homeassistant 7249604ba3ea[423]:     return {
Jul 28 19:38:33 homeassistant 7249604ba3ea[423]:   File "/usr/src/homeassistant/homeassistant/components/slack/notify.py", line 157, in <dictcomp>
Jul 28 19:38:33 homeassistant 7249604ba3ea[423]:     key: _async_templatize_blocks(hass, item) for key, item in value.items()
Jul 28 19:38:33 homeassistant 7249604ba3ea[423]:   File "/usr/src/homeassistant/homeassistant/components/slack/notify.py", line 154, in _async_templatize_blocks
Jul 28 19:38:33 homeassistant 7249604ba3ea[423]:     return [_async_templatize_blocks(hass, item) for item in value]
Jul 28 19:38:33 homeassistant 7249604ba3ea[423]:   File "/usr/src/homeassistant/homeassistant/components/slack/notify.py", line 154, in <listcomp>
Jul 28 19:38:33 homeassistant 7249604ba3ea[423]:     return [_async_templatize_blocks(hass, item) for item in value]
Jul 28 19:38:33 homeassistant 7249604ba3ea[423]:   File "/usr/src/homeassistant/homeassistant/components/slack/notify.py", line 156, in _async_templatize_blocks
Jul 28 19:38:33 homeassistant 7249604ba3ea[423]:     return {
Jul 28 19:38:33 homeassistant 7249604ba3ea[423]:   File "/usr/src/homeassistant/homeassistant/components/slack/notify.py", line 157, in <dictcomp>
Jul 28 19:38:33 homeassistant 7249604ba3ea[423]:     key: _async_templatize_blocks(hass, item) for key, item in value.items()
Jul 28 19:38:33 homeassistant 7249604ba3ea[423]:   File "/usr/src/homeassistant/homeassistant/components/slack/notify.py", line 156, in _async_templatize_blocks
Jul 28 19:38:33 homeassistant 7249604ba3ea[423]:     return {
Jul 28 19:38:33 homeassistant 7249604ba3ea[423]:   File "/usr/src/homeassistant/homeassistant/components/slack/notify.py", line 157, in <dictcomp>
Jul 28 19:38:33 homeassistant 7249604ba3ea[423]:     key: _async_templatize_blocks(hass, item) for key, item in value.items()
Jul 28 19:38:33 homeassistant 7249604ba3ea[423]:   File "/usr/src/homeassistant/homeassistant/components/slack/notify.py", line 160, in _async_templatize_blocks
Jul 28 19:38:33 homeassistant 7249604ba3ea[423]:     tmpl = template.Template(value, hass=hass)  # type: ignore  # no-untyped-call
Jul 28 19:38:33 homeassistant 7249604ba3ea[423]:   File "/usr/src/homeassistant/homeassistant/helpers/template.py", line 306, in __init__
Jul 28 19:38:33 homeassistant 7249604ba3ea[423]:     raise TypeError("Expected template to be a string")
Jul 28 19:38:33 homeassistant 7249604ba3ea[423]: TypeError: Expected template to be a string
```


### Additional information

_No response_

--- a/homeassistant/components/slack/notify.py
+++ b/homeassistant/components/slack/notify.py
@@ -146,21 +146,6 @@ def _async_sanitize_channel_names(channel_list: list[str]) -> list[str]:
     return [channel.lstrip("#") for channel in channel_list]
 
 
-@callback
-def _async_templatize_blocks(hass: HomeAssistant, value: Any) -> Any:
-    """Recursive template creator helper function."""
-    if isinstance(value, str):
-        tmpl = template.Template(value, hass=hass)  # type: ignore
-        return tmpl.async_render(parse_result=False)
-    if isinstance(value, list):
-        return [_async_templatize_blocks(hass, item) for item in value]
-    if isinstance(value, dict):
-        return {
-            key: _async_templatize_blocks(hass, item) for key, item in value.items()
-        }
-    return value
-
-
 class SlackNotificationService(BaseNotificationService):
     """Define the Slack notification logic."""
 
@@ -315,9 +300,9 @@ class SlackNotificationService(BaseNotificationService):
         # Message Type 1: A text-only message
         if ATTR_FILE not in data:
             if ATTR_BLOCKS_TEMPLATE in data:
-                blocks = _async_templatize_blocks(
-                    self._hass, data[ATTR_BLOCKS_TEMPLATE]
-                )
+                value = cv.template_complex(data[ATTR_BLOCKS_TEMPLATE])
+                template.attach(self._hass, value)
+                blocks = template.render_complex(value)
             elif ATTR_BLOCKS in data:
                 blocks = data[ATTR_BLOCKS]
             else:

--- a/homeassistant/components/slack/notify.py
+++ b/homeassistant/components/slack/notify.py
@@ -149,15 +149,16 @@ def _async_sanitize_channel_names(channel_list: list[str]) -> list[str]:
 @callback
 def _async_templatize_blocks(hass: HomeAssistant, value: Any) -> Any:
     """Recursive template creator helper function."""
+    if isinstance(value, str):
+        tmpl = template.Template(value, hass=hass)  # type: ignore
+        return tmpl.async_render(parse_result=False)
     if isinstance(value, list):
         return [_async_templatize_blocks(hass, item) for item in value]
     if isinstance(value, dict):
         return {
             key: _async_templatize_blocks(hass, item) for key, item in value.items()
         }
-
-    tmpl = template.Template(value, hass=hass)  # type: ignore  # no-untyped-call
-    return tmpl.async_render(parse_result=False)
+    return value
 
 
 class SlackNotificationService(BaseNotificationService):

